server/server.go

```diff
@@ -263,6 +265,54 @@ func (r *rodeServer) ListOccurrences(ctx context.Context, occurrenceRequest *pb.
 	}, nil
 }
 
+func (r *rodeServer) ValidatePolicy(ctx context.Context, policy *pb.ValidatePolicyRequest) (*pb.ValidatePolicyResponse, error) {
+	log := r.logger.Named("ValidatePolicy")
+
+	if len(policy.Policy) == 0 {
+		return nil, createError(log, "empty policy passed in", nil)
+	}
+
+	// Generate the AST
+	mod, err := ast.ParseModule("validate_module", string(policy.Policy))
+	if err != nil {
+		log.Debug("failed to parse the policy", zap.Any("policy", err))
+		return &pb.ValidatePolicyResponse{
+			Policy:  policy.Policy,
+			Compile: false,
+			Errors:  []string{err.Error()},
+		}, nil
+	}
+
+	// Create a new compiler instance and compile the module. Should this be moved to initialize function?
```

Comment:
Making a new one might be for the best? It look like there's a lot of internal state there that might end up being shared between runs.